File: leetcode3/변지협/v2/4024. Nearest Available Drone.py
class Solution:
    def nearestDrone(self, drones: list[list[int]], target: list[int]) -> int:
        lst = []
        tx,ty = target

        for x,y,r in drones:
            distance = abs(x-tx) + abs(y-ty)
            if distance <= r:
                lst.append(distance)
            else:
                lst.append(99999)
        
        _min = min(lst)
        
        if _min == 99999:
            return -1
        else:
            return lst.index(_min)

File: leetcode3/염혜정/4024. Nearest Available Drone.java
// O(3n)

class Solution {
    public int nearestDrone(int[][] drones, int[] target) {
        int x = target[0];
        int y = target[1];
        int min = Integer.MAX_VALUE;
        int idx = 0;
        for (int i = 0; i<drones.length; i++) {
            int md = Math.abs(drones[i][0]-x) + Math.abs(drones[i][1]-y);
            if (md > drones[i][2]) continue;
            if (md < min) {
                min = md;
                idx = i;
            }
        }
        if (min == Integer.MAX_VALUE) return -1;
        return idx;
    }
}
